src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php

```diff
@@ -247,42 +212,62 @@ public function getRedirectParameters($resource = null)
         $parameters = $redirect['parameters'];
 
         if (null !== $resource) {
-            $parameters = $this->parser->process($parameters, $resource);
+            $parameters = $this->parseResourceValues($parameters, $resource);
         }
 
         return $parameters;
     }
 
+    /**
+     * @return bool
+     */
     public function isLimited()
     {
-        return (bool) $this->parameters->get('limit', $this->settings['limit']);
+        return (bool) $this->parameters->get('limit', false);
     }
 
+    /**
+     * @return int|null
+     */
     public function getLimit()
     {
         $limit = null;
+
         if ($this->isLimited()) {
-            $limit = (int) $this->parameters->get('limit', $this->settings['limit']);
+            $limit = (int) $this->parameters->get('limit', 10);
         }
 
         return $limit;
     }
 
+    /**
+     * @return bool
+     */
     public function isPaginated()
     {
-        return (bool) $this->parameters->get('paginate', $this->settings['default_page_size']);
+        return (bool) $this->parameters->get('paginate', true);
     }
 
+    /**
+     * @return int
+     */
     public function getPaginationMaxPerPage()
     {
-        return (int) $this->parameters->get('paginate', $this->settings['default_page_size']);
+        return (int) $this->parameters->get('paginate', 10);
     }
 
+    /**
+     * @return bool
+     */
     public function isFilterable()
     {
-        return (bool) $this->parameters->get('filterable', $this->settings['filterable']);
+        return (bool) $this->parameters->get('filterable', false);
     }
 
+    /**
+     * @param array $criteria
+     * @return array
```

Comment:
Maybe not worry too much if https://github.com/Sylius/Sylius/pull/3784 is merged?